lib_vehicle_model/include/lib_vehicle_model/LibVehicleModel.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <memory>
+#include <stdexcept>
+#include <ros/ros.h>
+#include "ModelAccessException.h"
+#include "VehicleState.h"
+#include "VehicleMotionModel.h"
+#include "VehicleModelControlInput.h"
+#include "ParameterServer.h"
+#include "KinematicsSolver.h"
+#include "KinematicsProperty.h"
+
+namespace lib_vehicle_model {
+
+  //
+  // Public Namespace
+  //
+
+  /**
+   * @brief Initialization function for vehicle model. Loads the vehicle model as specified by parameters 
+   * 
+   * When provided with a NodeHandle this function uses the ROSParameterServer instead of a user specified parameter server
+   * 
+   * @param nh A pointer to the node handle which will be used to initialize the ROSParameterServer
+   * 
+   * @throws std::invalid_argument If the model could not be loaded or parameters could not be read
+   * @throws ModelAccessException If this function is called more than once within the same process execution
+   * 
+   */ 
+  void init(std::shared_ptr<ros::NodeHandle> nh);
+
+  /**
+   * @brief Initialization function for class. Loads the library as specified by parameters 
+   * 
+   * @param parameter_server A reference to the parameter server which vehicle models will use to load parameters
+   * 
+   * @throws std::invalid_argument If the model could not be loaded or parameters could not be read
+   * @throws ModelAccessException If this function is called more than once within the same process execution
+   * 
+   */ 
+  void init(std::shared_ptr<ParameterServer> parameter_server);
```

Comment:
Unit test

Reply:
You mean the existence of ParameterServer interface is purely for unit test purpose? If that is the case, I think create a MockNodeHandle class is more extensible...

Reply:
The parameter server interface is necessary to prevent vehicle models from becoming stateful by creating pub/sub setups. If we pass a node handle they would have access to the full ROS network instead of just read access to parameters. 

On a different node,
this is the NodeHandle header file. 
http://docs.ros.org/jade/api/roscpp/html/node__handle_8h_source.html
I'm not sure it is worth trying to mock this...

Reply:
Make sense to me.